src/components/dashboard/DashboardDetail.js

```diff
@@ -115,14 +115,14 @@ const mapDispatchToProps = {
 }
 
 // make sure that an action is an object, for UI purpose
-const postprocessSerdeStruct = (action) => typeof action == 'object' ? [action] : [{[action]: {}}]
+const postprocessSerdeStruct = (action) => typeof action == 'object' ? [{[action.kind]: [action.args]}] : [{[action]: {}}]
```

Comment:
I guess, this method is useless since we normalize actions with the DB, and there is only a single format now.